lib/database/models/climb_type/climb_type.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:drift/drift.dart';
+import 'package:redpoint/database/database.dart';
+import 'package:redpoint/database/models/grade/grade_system.dart';
+
+part 'climb_type.g.dart';
+
+class ClimbType extends Table {
+  IntColumn get id => integer()();
+  TextColumn get label => text()();
+
+  @override
+  Set<Column> get primaryKey => {id};
+}
+
+@DriftAccessor(tables: [ClimbType])
+class ClimbTypeDao extends DatabaseAccessor<AppDatabase> with _$ClimbTypeDaoMixin {
+  ClimbTypeDao(AppDatabase db) : super(db);
+
+  initializeData() { _init().forEach((element) async { await add(element); }); }
+
+  Future<List<ClimbTypeData>> get all => select(climbType).get();
+
+  Future<int> add(ClimbTypeCompanion entry) {
+    return into(climbType).insert(entry);
+  }
+}
+
+enum ClimbTypeEnum {
+  topRope(1, "Top Rope", [GradeSystemEnum.yds]),
+  boulder(2, "Boulder", [GradeSystemEnum.vScale]),
+  lead(3, "Lead", [GradeSystemEnum.yds]);
```

Comment:
Is this extensible in the future? Can other grade systems be used?

Reply:
Yes, I made the third property `final List<GradeSystemEnum> validGradeSystems;` a list so more valid grade systems could be added to the same type.